nano/secure/ledger_set_confirmed.cpp

```diff
@@ -0,0 +1,181 @@
+#include <nano/secure/ledger.hpp>
+#include <nano/secure/ledger_set_confirmed.hpp>
+#include <nano/store/account.hpp>
+#include <nano/store/component.hpp>
+#include <nano/store/confirmation_height.hpp>
+#include <nano/store/pending.hpp>
+#include <nano/store/pruned.hpp>
+
+nano::ledger_set_confirmed::ledger_set_confirmed (nano::ledger const & ledger) :
+	ledger{ ledger }
+{
+}
+
+std::optional<nano::uint128_t> nano::ledger_set_confirmed::balance (store::transaction const & transaction, nano::account const & account_a) const
+{
+	auto block = get (transaction, head (transaction, account_a));
+	if (!block)
+	{
+		return std::nullopt;
+	}
+	return block->balance ().number ();
+}
+
+// Balance for account containing hash
+std::optional<nano::uint128_t> nano::ledger_set_confirmed::balance (store::transaction const & transaction, nano::block_hash const & hash) const
```

Comment:
This seems to return balance for a specific block, not the latest value for account. Is this intentional?

Reply:
I removed the comment since the intent is to return the balance of the associated block by hash.

With the new block query functions we could potentially remove these convenience functions head/balance/height for block_hash and call sites can pull the value directly from the block.

Reply:
I think it is worth renaming it to `block_balance` to emphasise the fact that it behaves differently to the function definition (one line above) with the same name.